### ISSUE TITLE
How to create a reusable check-spelling action complete with configuration files

### ISSUE THREAD

**Issue by ricardo-dematos** [check-spelling/check-spelling#41]
I'm trying to create a wrapper to [check-spelling/check-spelling](check-spelling/check-spelling) , with the intention of  simplifying the inputs and outputs, while also providing a set of words valid in the [Valispace](https://www.valispace.com/) software context.

It resides in [valispace/actions/check-spelling](https://github.com/valispace/actions/tree/master/check-spelling), along with the [Configuration: Files](https://github.com/check-spelling/check-spelling/wiki/Configuration#files).

The main workflow has : 
 
```yml
- uses: valispace/actions/check-spelling@master
  with:
    event_aliases: '{ "schedule": "push" }'
    only_check_changed_files: '0'
```

The action is : 
 
```yml
runs:
  using: composite
  steps:
    - id: spelling
      uses: check-spelling/check-spelling@v0.0.21
      with:
        event_aliases: '${{ inputs.event_aliases }}'
        only_check_changed_files: '${{ inputs.only_check_changed_files }}'
        post_comment: '0'
        config: '.'
        extra_dictionaries:
          cspell:en_GB/src/wordsEnGb.txt
          cspell:en_GB/src/additional_words.txt
          cspell:en_US/src/en_US.txt
          cspell:en_US/src/additional_words.txt
          cspell:en_US/src/contractions-and-possessives.txt
          cspell:en_US/src/hyphenated-words.txt
          cspell:pt_PT/Portuguese-European.txt
          cspell:software-terms/src/software-terms.txt
          cspell:software-terms/src/software-tools.txt
          cspell:software-terms/src/network-protocols.txt
          cspell:software-terms/src/network-os.txt
          cspell:cryptocurrencies/cryptocurrencies.txt
          cspell:public-licenses/src/generated/public-licenses.txt
          cspell:public-licenses/src/additional-licenses.txt
          cspell:lorem-ipsum/dictionary.txt
          cspell:mnemonics/src/mnemonics.txt
          cspell:companies/src/companies.txt
          cspell:aws/aws.txt
          cspell:docker/src/docker-words.txt
          cspell:bash/bash-words.txt
          cspell:fullstack/fullstack.txt
          cspell:django/django.txt
          cspell:django/requirements.txt
          cspell:node/additional_words.txt
          cspell:node/node.txt
          cspell:node/src/node-old.txt
          cspell:npm/npm.txt
          cspell:python/src/python/python.txt
          cspell:python/src/python/python-lib.txt
          cspell:python/src/common/extra.txt
          cspell:python/src/additional_words.txt
          cspell:typescript/typescript.txt
          cspell:html/html.txt
          cspell:html-symbol-entities/entities.txt
          cspell:css/css.txt
          cspell:fonts/fonts.txt
          cspell:filetypes/filetypes.txt
```

This configuration does not read the [Configuration: Files](https://github.com/check-spelling/check-spelling/wiki/Configuration#files)!

I've tried in the [valispace/actions/check-spelling](https://github.com/valispace/actions/tree/master/check-spelling) action : 
* changing setting `config` to `'./check-spelling'`
* replacing setting `config` with just `spell_check_this: 'valispace/actions/check-spelling@master'`
* create symbolic links from `valispace/actions/check-spelling/.github/actions/spelling` to `valispace/actions/check-spelling` for the [Configuration: Files](https://github.com/check-spelling/check-spelling/wiki/Configuration#files)

None of this worked.

How can I make this work? :monocle_face: 

---

:point_right:  In logs it seems that [check-spelling/check-spelling](check-spelling/check-spelling) is only capturing [valispace/actions](https://github.com/valispace/actions) from setting `spell_check_this: 'valispace/actions/check-spelling@master'`.

**Comment by BenedekFarkas** [comment]
Hey @ricardo-dematos we've got both an action a workflow set up as a wrapper, please check out https://github.com/Lombiq/GitHub-Actions/blob/dev/Docs/Workflows.md#spelling-workflow!

**Comment by jsoref** [comment]
I'd generally add `debug: 1` and just read to see what it's doing.

Do you have an action run I could look through?

**Comment by BenedekFarkas** [comment]
@ricardo-dematos yes, so our https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/spelling/action.yml action is a wrapper for https://github.com/check-spelling/check-spelling/blob/main/action.yml with some extras to increase reusability of our base configuration in other repositories of ours:

- When the repository that we run the spell-checking on has its own configuration files, but not an advice.md, then we copy [advice.md from https://github.com/Lombiq/GitHub-Actions](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/spelling/advice.md) that is a link to our own set of guidelines for maintaining the spell-checking configuration of a given repository.
- Optionally merge/copy configuration files from https://github.com/Lombiq/GitHub-Actions to the repo under spell-checking, such as excludes.txt, line_forbidden.patterns and patterns.txt.
- [A few general-purpose dictionary files](https://github.com/Lombiq/GitHub-Actions/tree/dev/.github/actions/spelling/dictionaries) that you can optionally add to spell-checking.
- The ability to extend/override the general-purpose configuration source prefixes and list of dictionary files that we defined in the wrapper action.

For example, our [Open-Source Orchard Core Extensions](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions) repository runs spell-checking [in its Build and Test workflow](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/dev/.github/workflows/build-and-test.yml#L44) and even though it doesn't have its own patterns.txt file, it will "inherit" https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/spelling/patterns.txt.

Also, it has [an excludes.txt file](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/dev/.github/actions/spelling/excludes.txt) that will be extended with the entries from https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/spelling/excludes.txt.

**Comment by jsoref** [comment]
As of [v0.0.26](https://github.com/check-spelling/check-spelling/releases/v0.0.26), you can use [`load-config-from`](https://docs.check-spelling.dev/Configuration#load-config-from) with a list of fields and then put the corresponding configuration into [`config.json`](https://docs.check-spelling.dev/Configuration-Files#config) (obviously not including `config`) as opposed to storing configuration in the workflow. I'm not sure how helpful that'll be -- keep in mind that you can use [`config`](https://docs.check-spelling.dev/Configuration#config) to point to a path that doesn't live in the repository that's being checked (whose path you can set using [`experimental_path`](https://docs.check-spelling.dev/Configuration#experimentalpath)). If `expect.txt` is a symlink to a directory inside the repository, you could probably make something that works... 

It'd be something like:

```yaml
steps:
- uses: actions/checkout@...
  with:
    repository: owner/common-config
    path: check-spelling-config
- uses: actions/checkout@...
  with:
    path: code
- uses: check-spelling/check-spelling@v0.0.26
  with:
    config: config
    experimental_path: code
    load-config-from: |
      {
        "pr-trusted-keys": [
          "ignored",
          "errors",
          "notices",
          "warnings",
          ...
         ]
      }
```

---

I'm hoping that making the config `config.json` will make it easier for people to merge/inherit configuration. I'd love to hear if it works...
